pkg/maven/maven.go

```diff
@@ -162,6 +163,57 @@ func InstallMavenArtifacts(options *EvaluateOptions, utils Utils) error {
 	return doInstallMavenArtifacts(options, utils)
 }
 
+func FindArtifacts(options *EvaluateOptions, utils Utils) (piperenv.Artifacts, error) {
+	var artifacts piperenv.Artifacts
+
+	pomFiles, err := utils.Glob(filepath.Join("**", "pom.xml"))
+	if err != nil {
+		return nil, err
+	}
+
+	for _, pomFile := range pomFiles {
+		dir := filepath.Dir(pomFile)
+
+		finalName, err := Evaluate(options, "project.build.finalName", utils)
```

Comment:
What about having a look for `*.jar`, etc instead of doing separate maven calls?

Reply:
The output artifact might be created with the `.war` packaging, also using globs for artifacts search is potentially dangerous and might lead to unwanted files being treated as artifacts (like classes-jar file).
IMHO it's more precise and safer option to receive information using maven options evaluation.